src/pages/authentication/Login.jsx

```diff
@@ -1,51 +1,116 @@
-import "./authentication.css";
-import { Navbar } from "../../components";
+import "./Auth.css";
+import { Link, useNavigate } from "react-router-dom";
+import { TextInput, PasswordInput } from "./components";
+import { useState } from "react";
+import { useAuth } from "../../context";
+import axios from "axios";
 
 export default function Login() {
+  const [userData, setUserData] = useState({
+    email: "",
+    password: "",
+  });
+
+  const navigate = useNavigate();
+  const { authDispatch } = useAuth();
+
+  const doLoginNetworkCall = async () => {
+    try {
+      const response = await axios.post("/api/auth/login", {
+        email: userData.email,
+        password: userData.password,
+      });
+
+      switch (response.status) {
+        case 200:
+          localStorage.setItem("token", response.data.encodedToken);
+          authDispatch({
+            type: "LOGIN",
+            payload: {
+              user: response.data.foundUser,
+              token: response.data.encodedToken,
+            },
+          });
+          alert("Login Successfull!")
+          navigate("/");
+          break;
+        case 404:
+          throw new Error("Invalid Email ID");
+        case 401:
+          throw new Error("Incorrect password");
+        case 500:
+          throw new Error("Error occured while getting response from server");
+      }
+    } catch (error) {
+      alert(
+        "Error Occured: please check credentials or try again later.",
+        error
+      );
+    }
+  };
+
+  const handleLoginClick = (e) => {
+    e.preventDefault();
+
+    if (userData.email === "" && userData.password === "") {
+      alert("please enter appropriate fields");
```

Comment:
The message could be more descriptive. Something like 
`Email and Password cannot be empty`